Source/WebGPU/WebGPU/ComputePipeline.mm

```diff
@@ -39,8 +39,8 @@
 {
     auto computePipelineDescriptor = [MTLComputePipelineDescriptor new];
     computePipelineDescriptor.computeFunction = function;
-    // FIXME: check this calculation for overflow
-    computePipelineDescriptor.maxTotalThreadsPerThreadgroup = size.width * size.height * size.depth;
+    // -[MTLComputePipelineDescriptor setMaxTotalThreadsPerThreadgroup:] will release assert if maxTotalThreadsPerThreadgroup > USHRT_MAX
+    computePipelineDescriptor.maxTotalThreadsPerThreadgroup = std::min<NSUInteger>(size.width * size.height * size.depth, USHRT_MAX);
```

Comment:
Instead of clamping, should this be an error?

Reply:
Initially I made it an error but based on the Metal documentation I don't think its correct to be an error.

I think on iOS (and Apple Silicon), values above 1024 will be treated as 1024. But anything above USHRT_MAX will cause a crash in Metal.framework.

I will add a FIXME to investigate further. The goal of this patch is to avoid a crash while running the CTS.

Reply:
https://bugs.webkit.org/show_bug.cgi?id=265910

Will add a FIXME here